### PR TITLE
refactor(router): code cleaning

### PR DESCRIPTION
**Pull request by JeanMeche**
Removing unnecessary type assertions (`as any`),  non-null assertions etc.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No

### CODE REVIEW

File: goldens/public-api/router/index.md [hunk 1]
```diff
@@ -793,7 +793,7 @@ export class RouterLinkActive implements OnChanges, OnDestroy, AfterContentInit
     constructor(router: Router, element: ElementRef, renderer: Renderer2, cdr: ChangeDetectorRef, link?: RouterLink | undefined);
     ariaCurrentWhenActive?: 'page' | 'step' | 'location' | 'date' | 'time' | true | false;
     // (undocumented)
-    readonly isActive: boolean;
+    get isActive(): boolean;
```
Comment (JeanMeche) [inline goldens/public-api/router/index.md:796]: I'm wondering if that would be considered a breaking change or not ?

File: packages/router/src/directives/router_link_active.ts [hunk 2]
```diff
@@ -97,7 +97,11 @@ export class RouterLinkActive implements OnChanges, OnDestroy, AfterContentInit
   private classes: string[] = [];
   private routerEventsSubscription: Subscription;
   private linkInputChangesSubscription?: Subscription;
-  public readonly isActive: boolean = false;
+  private _isActive: boolean = false;
```
Comment (atscott) [inline packages/router/src/directives/router_link_active.ts:100]:
```suggestion
  private _isActive = false;
```
nit: `: boolean` is not necessary with the initializer

File: goldens/public-api/router/index.md (hunk 1 above)
Comment (atscott) [reply goldens/public-api/router/index.md:796]: This should be fine I think.

File: packages/router/src/directives/router_link_active.ts [hunk 3]
```diff
@@ -97,7 +97,11 @@ export class RouterLinkActive implements OnChanges, OnDestroy, AfterContentInit
...
+
+  public get isActive() {
```
Comment (atscott) [inline packages/router/src/directives/router_link_active.ts:102]:
```suggestion
  get isActive() {
```
nit: `public` specifier not necessary

**Review by atscott — APPROVED** [review]
reviewed-for: public-api, fw-router

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
Caretaker note: it looks like this PR has the necessary set of approvals, so I'm adding the "merge" label.

**Comment by jessicajaniuk** [comment]
@JeanMeche This PR has merge conflicts with `15.0.x`. I'll update to `target: minor`. Please create a patch PR targeting `15.0.x` if you'd like this to land there.

**Pull request closed without merging**